Comp/src/main/java/frc/team4276/frc2024/Logging/LoggableRobotFile.java

```diff
@@ -131,4 +143,113 @@ public synchronized boolean writeToFile(String str, DebugLevel level ) {
     public String getFilePath() {
         return logPath;
     }
-}
+//helper functions
+    private static void checkAndReduceDirectorySize(Path dir) throws IOException {
+    if(getDirectorySizeInMB(dir) >= DebugConstants.maxDirSize){
+        while (getDirectorySizeInMB(dir) > DebugConstants.maxDirSize - DebugConstants.reductionSize && !areAllFilesNotRlog(dir)) {
+
+            String oldestFilePath = findOldestRlogFile(dir);
+            //dont want to delete files being used by other threads
+            if(isFileInUse(Paths.get(oldestFilePath)) || !validRlog(oldestFilePath)){
+                continue;
+            }
+            Files.delete(Paths.get(oldestFilePath));
+        }
+    }
+    
+    }
+
+    private static double getDirectorySizeInMB(Path dir) throws IOException {
+        final long[] totalSize = {0};
+
+        Files.walkFileTree(dir, new SimpleFileVisitor<Path>() {
+            @Override
+            public FileVisitResult visitFile(Path file, BasicFileAttributes attrs) throws IOException {
+                if (Files.isRegularFile(file)) {
```

Comment:
Fine to assume in this case that there are no subdirectories, but if you wanted to make this more generic you could make a recursive call to this same function when the file is a directory